docs/client.md

```diff
@@ -0,0 +1,53 @@
+# pb-cli specification
+
+## Overview
+
+`pb-cli` is a command line tool which communicate with `pb-go` backend. Although user can just upload data to pastebin directly with `curl`, we still provide a tools to help user easily use our advanced feature such as burn-after-reading and private share.
+
+We guarantee everything that worked with `pb-cli`, also worked with `curl`.
+
+## Configuration
+
+You can use `-c file` to provide a config file. If you didn't provide one, it will find config file called `.pb-cli.toml` from current directory then home directory.
+
+Config file should be written in `toml` like:
+
+```toml
+host = "http://your.pastebin.bakcend:port"
+master_key = "Same as your config on server"
+```
+
+You can use `-h <url>` with any command in `pb-cli`, it is a global flag.
+You can use `-k <master_key>` with `delete` command.
+
+Using flag in commands will override the value from config file.
+
+## Upload
+
+Usage:
+
+```text
+pb-cli upload [options] <file>
+-p                Optional. Private share. Will using a random password for private share.
+-P <PASSWORD>     Optional. Private share with specificated password.
+-e <EXPIRE>       Optional. Set to 0 means burn-after-read. Default 24. (unit: hrs)
+```
+
+When file is not given, `pb-cli` will read data from `stdin`, like as a pipe.
+
```

Comment:
Please do remember, even user offers nothing about the optional flag, the param in url should exist even so.

The default expire hour should be extract from `/status` (HTTP GET) api. It's not a static value. Just hold it here, I will try to build a doc about `/status` later corresponding to its response.

Reply:
The status api code is already pushed into upstream, check: 

04151a6 (https://github.com/pb-go/pb-go/commit/04151a6ccf7635efb56f3778209d9edb080aa9d0): feat(statusapi): make status api to be json and serialized

Reply:
👍

Reply:
Note: When got a response from server for default expire, the expire setting validity should be checked on the client side first, although we already have server side validation for safety guarantee.